tests/unit/amo/reducers/testReviews.js

```diff
@@ -124,6 +132,32 @@ describe(__filename, () => {
     expect(state.byUserId[userId]).not.toBeDefined();
   });
 
+  it('resets the byUserId data when adding an internal review', () => {
```

Comment:
When you factor out a shared piece of logic (e.g. `addReviewInState()`) you have two options:
1. Duplicate the tests to cover the shared logic in every place you depend on it
2. Add unit tests for the new shared function and use a sinon mock to *guarantee* the function is getting called correctly

You have taken the duplication approach here but I'd rather see you move to the second option.

First, you'll need to set it up for dependency injection:

```diff
diff --git a/src/amo/reducers/reviews.js b/src/amo/reducers/reviews.js
index 0b629a179..b11445a46 100644
--- a/src/amo/reducers/reviews.js
+++ b/src/amo/reducers/reviews.js
@@ -230,7 +230,7 @@ export const selectLatestUserReview = ({
 };
 
 // This is used in the SET_REVIEW and SET_INTERNAL_REVIEW cases.
-const addReviewInState = ({
+export const addReviewInState = ({
   state,
   review,
 }: {|
@@ -266,6 +266,9 @@ type ReviewActionType =
 export default function reviewsReducer(
   state: ReviewsState = initialState,
   action: ReviewActionType,
+  {
+    _addReviewInState = addReviewInState,
+  }: {| _addReviewInState: typeof addReviewInState |} = {},
 ) {
   switch (action.type) {
     case SEND_REPLY_TO_REVIEW:
@@ -337,12 +340,12 @@ export default function reviewsReducer(
       const { payload } = action;
       const review = createInternalReview(payload);
 
-      return addReviewInState({ state, review });
+      return _addReviewInState({ state, review });
     }
     case SET_INTERNAL_REVIEW: {
       const { payload } = action;
 
-      return addReviewInState({ state, review: payload });
+      return _addReviewInState({ state, review: payload });
     }
     case SET_REVIEW_REPLY: {
       const reviewId = action.payload.originalReviewId;
```

Next, add unit tests for `addReviewInState()` and give yourself two test cases for `SET_REVIEW` and `SET_INTERNAL_REVIEW` with mock functions to make sure it's used.

Reply:
😞